server/src/main/java/org/elasticsearch/action/admin/indices/forcemerge/TransportForceMergeAction.java

```diff
@@ -77,10 +80,16 @@ protected ForceMergeRequest readRequestFrom(StreamInput in) throws IOException {
 
     @Override
     protected void shardOperation(ForceMergeRequest request, ShardRouting shardRouting, ActionListener<EmptyResult> listener) throws IOException {
-        IndexShard indexShard = indicesService.indexServiceSafe(shardRouting.shardId().getIndex()).getShard(shardRouting.shardId().id());
-        indexShard.flush(false);
-        indexShard.forceMerge(request);
-        listener.onResponse(EmptyResult.INSTANCE);
+        threadPool.executor(ThreadPool.Names.FORCE_MERGE).execute(() -> {
+            try {
+                IndexShard indexShard = indicesService.indexServiceSafe(shardRouting.shardId().getIndex()).getShard(shardRouting.shardId().id());
+                indexShard.flush(false);
+                indexShard.forceMerge(request);
+                listener.onResponse(EmptyResult.INSTANCE);
+            } catch (Exception e) {
+                listener.onFailure(e);
+            }
+        });
```

Comment:
I think this also ought to handle RejectedExecutionException somehow?

Reply:
Could rethrow to avoid stuck operations in case we make it bounded in the future for whatever reason.

Reply:
Oh nevermind - the `shardOperation` itself is already in a try/catch block. All good